sklearn/base.py

```diff
@@ -290,10 +290,15 @@ def __repr__(self):
                                                offset=len(class_name),),)
 
     def __getstate__(self):
+        try:
+            state = super(BaseEstimator, self).__getstate__()
+        except AttributeError:
+            state = self.__dict__
+
         if type(self).__module__.startswith('sklearn.'):
```

Comment:
This conditional is not properly covered, because the classes I use in the test are from within the sklearn module namespace (since scikit-learn has its tests in the sklearn namespace). So basically `return state.copy()` is never reached in the tests. Problem is, I cannot patch `type(self).__module__` without breaking the pickling mechanism (it makes a lookup for the module). So either we make the string `'sklearn.'` in the `BaseEstimator` mock-patchable, or we need to create a test-class outside of the sklearn namespace, to get this conditional fully covered.